pdal_ign_macro/main_preprocessing_mnx.py

```diff
@@ -0,0 +1,138 @@
+    )
+    parser.add_argument(
+        "--keep_temporary_dims",
+        "-k",
+        action="store_true",
+        help="If set, do not delete temporary dimensions",
+    )
+    parser.add_argument(
+        "--skip_buffer",
+        action="store_true",
+        help="If set, skip adding a buffer from the neighbor tiles based on their name",
+    )
+    parser.add_argument(
+        "--buffer_width",
+        type=float,
+        default=25,
+        help="width of the border to add to the tile (in meters)",
+    )
+    parser.add_argument(
+        "--spatial_ref",
+        type=str,
+        required=True,
+        help="spatial reference for the writer",
+    )
+    parser.add_argument(
+        "--virtual_points_classes",
+        "-c",
+        type=int,
+        default=66,
+        help="classification value to assign to the added virtual points",
+    )
+    parser.add_argument(
+        "--tile_width",
+        type=int,
+        default=1000,
+        help="width of tiles in meters (required when running with a buffer)",
+    )
+    parser.add_argument(
+        "--tile_coord_scale",
+        type=int,
+        default=1000,
+        help="scale used in the filename to describe coordinates in meters (required when running with a buffer)",
+    )
+    parser.add_argument("ARGUMENTS ADD POINTS")
+
+
+def preprocess_mnx(
+    input_las: str,
+    input_geojson: str,
+    output_las: str,
+    dsm_dimension: str,
+    dtm_dimension: str,
+    output_dsm: str,
+    output_dtm: str,
+    keep_temporary_dims: bool,
+    skip_buffer: bool,
+    buffer_width: float,
+    spatial_ref: str,
+    virtual_points_classes: int,
+    tile_width: int,
+    tile_coord_scale: int,
+):
+    """Lauch preprocessing before calculating MNX
+    Args:
+        input_las (str): Path to the LIDAR `.las/.laz` file.
+        input_geosjon (str): Path to the input GeoJSON file with 3D points.
+        output_las (str): Path to save the updated LIDAR file (LAS/LAZ format).
+        dsm_dimension (str): Dimension name for the output DSM marker
+        dtm_dimension (str): Dimension name for the output DTM marker
+        output_dsm (str): utput dsm tiff file
+        output_dtm (str): utput dtm tiff file
+        keep_temporary_dims (bool): If set, do not delete temporary dimensions
+        skip_buffer (bool): If set, skip adding a buffer from the neighbor tiles based on their name
+        buffer_width (float): width of the border to add to the tile (in meters)
+        spatial_ref (str): CRS's value of the data in 'EPSG:XXXX' format
+        virtual_points_classes (int):  The classification value to assign to those virtual points (default: 66).
+        tile_width (int): Width of the tile in meters (default: 1000).
+        tile_coord_scale (int): scale used in the filename to describe coordinates in meters (default: 1000).
+    """
+    with tempfile.NamedTemporaryFile(suffix="_intermediate.las", dir=".") as tmp_las:
```

Comment:
pour avoir un fichier plus léger : 
```suggestion
    with tempfile.NamedTemporaryFile(suffix="_intermediate.laz", dir=".") as tmp_las:
```